lib/safe-http-client.js

```diff
@@ -0,0 +1,134 @@
+import dns from 'node:dns/promises'
+
+import {fetch} from 'undici'
+import ipaddr from 'ipaddr.js'
+
+import {IMAGE_MIME_TYPES} from './constants.js'
+
+export class HttpError extends Error {
+  /**
+   * @param {number} statusCode
+   * @param {string} message
+   */
```

Comment:
```suggestion
  /**
   * Create an HTTP error.
   * 
   * @param {number} statusCode
   *   Status code.
   * @param {string} message
   *   Message.
   * @returns
   *   HTTP error.
   */
```

Not supor useful, but nice to have some docs.